electrum/interface.py

```diff
@@ -362,6 +367,8 @@ def _is_saved_ssl_cert_available(self):
         with open(self.cert_path, 'r') as f:
             contents = f.read()
         if contents == '':  # CA signed
+            if self.network.config.get("serverfingerprint"):
+                raise InvalidOptionCombination("cannot use --serverfingerprint with CA signed servers")
```

Comment:
As above, this should also test for `is_main_server`.

How about introducing a new method, `interface._get_expected_fingerprint` or similar, which returns
`self.network.config.get("serverfingerprint")` for the main server, and None otherwise? That way, the `is_main_server` tests don't have to be duplicated.